[{"client_msg_id":"f5045a19-5981-4aec-8540-d8d97208a4d3","type":"message","text":"Anyone familiar with PyCall? I have a config issue that is being difficult to debug. Somehow building PyCall finds the wrong libpython","user":"U7LNECWEA","ts":"1616764051.457800","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"35qgB","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"Anyone familiar with PyCall? I have a config issue that is being difficult to debug. Somehow building PyCall finds the wrong libpython"}]}]}],"thread_ts":"1616764051.457800","reply_count":6,"reply_users_count":2,"latest_reply":"1616765025.459300","reply_users":["U7LNECWEA","U01GXNFKY6R"],"is_locked":false,"subscribed":false},{"client_msg_id":"f6c6ef34-f2dc-45fb-96da-e043363182e6","type":"message","text":"```[nibbler deps]% pwd\n/Users/cmcbride/.julia/packages/PyCall/tqyST/deps\n[nibbler deps]% which python\n/Users/cmcbride/.pyenv/shims/python\n[nibbler deps]% python --version\nPython 3.8.7\n[nibbler deps]% python find_libpython.py\n/Users/cmcbride/.pyenv/versions/3.8.7/lib/python3.8/config-3.8-darwin/libpython3.8.a%\n[nibbler deps]% /usr/local/bin/python find_libpython.py\n/usr/local/Cellar/python@2/2.7.17_1/Frameworks/Python.framework/Versions/2.7/Python%\n[nibbler deps]% cat deps.jl\nconst python = \"/Users/cmcbride/.pyenv/versions/3.8.7/bin/python\"\nconst libpython = \"/System/Library/Frameworks/Python.framework/Versions/2.7/Python\"\nconst pyprogramname = \"/Users/cmcbride/.pyenv/versions/3.8.7/bin/python\"\nconst pyversion_build = v\"3.8.7\"\nconst PYTHONHOME = \"/Users/cmcbride/.pyenv/versions/3.8.7:/Users/cmcbride/.pyenv/versions/3.8.7\"\n\n\"True if we are using the Python distribution in the Conda package.\"\nconst conda = false```","user":"U7LNECWEA","ts":"1616764071.457900","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"qjO4","elements":[{"type":"rich_text_preformatted","elements":[{"type":"text","text":"[nibbler deps]% pwd\n/Users/cmcbride/.julia/packages/PyCall/tqyST/deps\n[nibbler deps]% which python\n/Users/cmcbride/.pyenv/shims/python\n[nibbler deps]% python --version\nPython 3.8.7\n[nibbler deps]% python find_libpython.py\n/Users/cmcbride/.pyenv/versions/3.8.7/lib/python3.8/config-3.8-darwin/libpython3.8.a%\n[nibbler deps]% /usr/local/bin/python find_libpython.py\n/usr/local/Cellar/python@2/2.7.17_1/Frameworks/Python.framework/Versions/2.7/Python%\n[nibbler deps]% cat deps.jl\nconst python = \"/Users/cmcbride/.pyenv/versions/3.8.7/bin/python\"\nconst libpython = \"/System/Library/Frameworks/Python.framework/Versions/2.7/Python\"\nconst pyprogramname = \"/Users/cmcbride/.pyenv/versions/3.8.7/bin/python\"\nconst pyversion_build = v\"3.8.7\"\nconst PYTHONHOME = \"/Users/cmcbride/.pyenv/versions/3.8.7:/Users/cmcbride/.pyenv/versions/3.8.7\"\n\n\"True if we are using the Python distribution in the Conda package.\"\nconst conda = false"}]}]}],"thread_ts":"1616764051.457800","parent_user_id":"U7LNECWEA"},{"client_msg_id":"9cbd7dd3-eb8a-4c12-9a9b-ed26bda5061b","type":"message","text":"in the README it shows you how to build PyCall by specifying a path to python","user":"U01GXNFKY6R","ts":"1616764562.458400","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"J8W","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"in the README it shows you how to build PyCall by specifying a path to python"}]}]}],"thread_ts":"1616764051.457800","parent_user_id":"U7LNECWEA"},{"client_msg_id":"c3cd601e-fe05-4dde-9c0b-b49c7a245c2d","type":"message","text":"yeah, I am just showing how it was inconsistently picking python3 and then a libpython that was associated with the system.","user":"U7LNECWEA","ts":"1616764607.458600","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"YTdL","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"yeah, I am just showing how it was inconsistently picking python3 and then a libpython that was associated with the system."}]}]}],"thread_ts":"1616764051.457800","parent_user_id":"U7LNECWEA"},{"client_msg_id":"350b2fe0-b76b-4c4d-b9db-303e2f9b82ac","type":"message","text":"I think this resolved it: <https://github.com/JuliaPy/PyCall.jl/issues/597>","user":"U7LNECWEA","ts":"1616764638.458800","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"ZJZoL","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"I think this resolved it: "},{"type":"link","url":"https://github.com/JuliaPy/PyCall.jl/issues/597"}]}]}],"thread_ts":"1616764051.457800","parent_user_id":"U7LNECWEA"},{"client_msg_id":"080b02cc-8977-4f51-94f9-dc2b3a0fc5db","type":"message","text":"so, PyCall found the right python executable but because there wasn’t a shared library that matched, it found a different library (from the system) which made it inconsistent.","user":"U7LNECWEA","ts":"1616764807.459000","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"Uj/","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"so, PyCall found the right python executable but because there wasn’t a shared library that matched, it found a different library (from the system) which made it inconsistent."}]}]}],"thread_ts":"1616764051.457800","parent_user_id":"U7LNECWEA"},{"client_msg_id":"1d9df3a5-b1e0-49da-8eb1-dc5df236231d","type":"message","text":"ah okay! that’s different from what I had thought","user":"U01GXNFKY6R","ts":"1616765025.459300","team":"T68168MUP","blocks":[{"type":"rich_text","block_id":"UV/o","elements":[{"type":"rich_text_section","elements":[{"type":"text","text":"ah okay! that’s different from what I had thought"}]}]}],"thread_ts":"1616764051.457800","parent_user_id":"U7LNECWEA"}]